api/preprints/views.py

```diff
@@ -107,6 +108,7 @@ class PreprintDetail(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView, Pre
     permission_classes = (
         drf_permissions.IsAuthenticatedOrReadOnly,
         base_permissions.TokenHasScope,
+        ModeratorIfNeverPublicWithdrawn,
```

Comment:
[404](https://github.com/CenterForOpenScience/osf.io/pull/8586/files#diff-6cfd1732a416f15d2594c72d89c0f8ffR54), because viewing preprints normally doesn't require auth, and no one can see deleted preprints.